graph/src/main/scala/net/spals/dsa/graph/Graph.scala

```diff
@@ -30,18 +30,18 @@ case class WeightedGraph[T, W](vertices: Set[Vertex[T]], edgeFunc: (Vertex[T], V
     WeightedGraph(vertices + v, edgeFunc)
   }
 
-  def getEdge(v1: Vertex[T], v2: Vertex[T]): Option[W] = {
-    edgeFunc.apply(v1, v2)
+  def getEdge(v1: Vertex[T], v2: Vertex[T]): Option[Edge[T, W]] = {
+    edgeFunc.apply(v1, v2).map(Edge(v1, v2, _))
   }
 
   def getEdges(v1: Vertex[T]): Set[Edge[T, W]] = {
-    vertices.map(v2 => (v2, edgeFunc.apply(v1, v2)))
+    vertices.map(v2 => (v2, edgeFunc.apply(v1, v2) /*weight*/))
```

Comment:
Just to remind myself what the edge function is producing in the second part of the tuple